### PR TITLE
[GITHUB-1] Prometheus Nginx Exporter

### PR DESCRIPTION
**Pull request by rsgoodall**

### CODE REVIEW

File: tasks/build.yml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  lineinfile:
+    dest: /etc/default/prometheus-nginx-exporter
+    state: present
+    create: yes
+    line: "START=yes"
+
+- stat: path=/usr/local/bin/gosu
+  register: gosu_bin
+
+- name: Download gosu
```
Comment (racer1680) [inline tasks/build.yml]: Why do you need it?

**Comment by lobsterdore** [comment]
@quater @wojtek-oledzki if you have some spare time :)

File: defaults/main.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+---
+
+prometheus_nginx_exporter:
+  checksum: sha256:7cf6a4c1d29fb42f43e05b164ca6c2fdeeaa81484112fc312f3dc7fc22edfe35
+  # environment: dev
```
Comment (lobsterdore) [inline defaults/main.yml]: Remove this line if the var isn't needed.

File: tasks/build.yml [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    owner: "{{ prometheus_nginx_exporter.user }}"
+    group: "{{ prometheus_nginx_exporter.group }}"
+    mode: 0750
+  with_items:
+    - "{{ prometheus_nginx_exporter.path.install }}"
+    - "{{ prometheus_nginx_exporter.path.log }}"
+
+- stat:
+    path: "/tmp/nginx_exporter_v{{ prometheus_nginx_exporter.version }}_linux_amd{{ ansible_userspace_bits }}.tar.gz"
+  register: exporter_tar
```
Comment (lobsterdore) [inline tasks/build.yml]: I know lint went funny for you but can you try an empty line here again?

File: tasks/build.yml [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+- name: Untar Prometheus Nginx Exporter tarball
+  become: yes
+  unarchive:
+    src: "/tmp/nginx_exporter_v{{ prometheus_nginx_exporter.version }}_linux_amd{{ ansible_userspace_bits }}.tar.gz"
+    dest: "{{ prometheus_nginx_exporter.path.install }}/prometheus_nginx_exporter"
+    owner: "{{ prometheus_nginx_exporter.user }}"
+    group: "{{ prometheus_nginx_exporter.group }}"
+    remote_src: True
+
+# Ansible 2.3.x use path: /etc/default/prometheus
```
Comment (lobsterdore) [inline tasks/build.yml]: Is this comment relevant?

File: meta/main.yml [hunk 5]
```diff
@@ -0,0 +1,22 @@
+---
+
+galaxy_info:
+  author: "Rob Goodall"
```
Comment (racer1680) [inline meta/main.yml]: We don't need that

File: tasks/build.yml (hunk 1 above)
Comment (lobsterdore) [inline tasks/build.yml]: Do we definitely need gosu?

File: templates/prometheus_nginx_exporter.sysvinit.sh.j2 [hunk 6]
```diff
@@ -0,0 +1,141 @@
+#!/bin/sh
```
Comment (racer1680) [inline templates/prometheus_nginx_exporter.sysvinit.sh.j2]: Do we still add .sh at the end of bash file?

Comment (racer1680) [inline templates/prometheus_nginx_exporter.sysvinit.sh.j2]: /bin/bash instead ?

**Comment by racer1680** [comment]
Any chance we could have short description in README what this does?

**Comment by lobsterdore** [comment]
Good point @racer1680, there is a basic template here -> https://github.com/sansible/sansible/blob/scaffold/README.md

File: defaults/main.yml [hunk 7]
```diff
@@ -0,0 +1,13 @@
+---
+
+prometheus_nginx_exporter:
+  start_on_boot: yes
```
Comment (lobsterdore) [inline defaults/main.yml]: Can you alphabetically order this?

File: tasks/build.yml [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+
+- name: Make Prometheus Nginx Exporter service start on boot
+  become: yes
+  service:
+    name: prometheus-nginx-exporter
+    enabled: yes
+    state: started
+  when: prometheus_nginx_exporter.start_on_boot
+
+- name: Start Exporter
```
Comment (lobsterdore) [inline tasks/build.yml]: I don't think we need this second start task

**Review by lobsterdore — COMMENTED** [review]
LGTM 👍

File: tasks/build.yml [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+    mode: 0755
+
+- stat:
+    path: "/tmp/nginx_exporter_v{{ prometheus_nginx_exporter.version }}_linux_amd{{ ansible_userspace_bits }}.tar.gz"
+  register: exporter_tar
+
+- name: Download Prometheus Nginx Exporter tarball
+  become: yes
+  get_url:
+    url: "https://github.com/c4milo/nginx_exporter/releases/download/v{{ prometheus_nginx_exporter.version }}/nginx_exporter_v{{ prometheus_nginx_exporter.version }}_linux_amd{{ ansible_userspace_bits }}.tar.gz"
```
Comment (racer1680) [inline tasks/build.yml]: I think we usually put URL into defaults/main.yml otherwise LGTM

File: tests/Vagrantfile [hunk 10]
```diff
@@ -0,0 +1,52 @@
+# -*- mode: ruby -*-
```
Comment (racer1680) [inline tests/Vagrantfile:1]: VM doesn't have private IP. Is that intentional?

File: tasks/build.yml [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+  register: exporter_tar
+
+- name: Download Prometheus Nginx Exporter tarball
+  become: yes
+  get_url:
+    url: "{{ prometheus_nginx_exporter.download_url }}"
+    dest: "/tmp"
+    checksum: "{{ prometheus_nginx_exporter.checksum }}"
+  when: exporter_tar.stat.exists == False
+  ignore_errors: yes
```
Comment (racer1680) [inline tasks/build.yml]: Are you sure?

File: tests/Vagrantfile (hunk 10 above)
Comment (lobsterdore) [reply tests/Vagrantfile:1]: You only need a private IP if you are going to assign a hostname to the machine.

Comment (lobsterdore) [reply tests/Vagrantfile:1]: Or if you're using NFS :)

Comment (racer1680) [reply tests/Vagrantfile:1]: :)

File: tasks/build.yml [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+  become: yes
+  copy:
+    src: prometheus_exporter.sh
+    dest: "{{ prometheus_nginx_exporter.path.install }}/prometheus_exporter.sh"
+    owner: "{{ prometheus_nginx_exporter.user }}"
+    group: "{{ prometheus_nginx_exporter.group }}"
+    mode: 0755
+
+- stat:
+    path: "/tmp/nginx_exporter_v{{ prometheus_nginx_exporter.version }}_linux_amd{{ ansible_userspace_bits }}.tar.gz"
```
Comment (racer1680) [inline tasks/build.yml]: Wouldn't be better to control name of the file as variable in defaults?

Comment (lobsterdore) [reply tasks/build.yml]: Perhaps it should be taken from the URL.

**Comment by racer1680** [comment]
LGTM

**Review by lobsterdore — APPROVED** [review]
LGTM 👍

**Pull request merged**